### PR TITLE
Refactor dataset_test.py

### LINKED ISSUE #914: Refactor and reenable or remove Python tests for Dataset
`python/tests/dataset.py` contains a large number of commented tests that have been disabled and not refactored last May when `Dataset` was rewritten.

- Find the useful tests (things that are not covered elsewhere).
- Refactor them.
- Remove all others.

### PR DESCRIPTION
**Pull request by OwenArnold**
Better relevant test coverage. Remove dead code, we still skip some tests. `test_data_array.py` has also been enhanced as part of this 

Fixes #914

Opened new issue #1085 specifically to treat slicing operations for DataArray and Dataset

### CODE REVIEW

File: python/tests/test_dataset.py [hunk 1]
```diff
@@ -8,9 +8,23 @@
...
 
 
+def test_shape():
+    a = sc.Variable(value=1)
+    d = sc.Dataset(data={'a': a})
+    assert d.shape == []
+    a = sc.Variable(['x'], shape=[2])
+    b = sc.Variable(['y', 'z'], shape=[3, 4])
+    d = sc.Dataset(data={'a': a, 'b': b})
+    assert d.shape == [4, 3, 2]
```
Comment (SimonHeybrock) [inline python/tests/test_dataset.py:18]: Are we making any promises about dimension order in `Dataset`? In principle it is undefined.

File: python/tests/test_dataset.py [hunk 2]
```diff
@@ -23,13 +37,15 @@ def test_create():
     assert d.coords['y'] == y
     assert d['xy'].data == xy
     assert d['x'].data == x
+    assert d.dims == ['y', 'x']
```
Comment (SimonHeybrock) [inline python/tests/test_dataset.py:40]: See above.

File: python/tests/test_dataset.py [hunk 3]
```diff
@@ -23,13 +37,15 @@ def test_create():
...
 
 
 def test_create_from_data_array():
     var = sc.Variable(dims=['x'], values=np.arange(4))
     base = sc.Dataset({'a': var}, coords={'x': var, 'aux': var})
     d = sc.Dataset(base['a'])
     assert d == base
+    assert base['a'] == sc.DataArray(data=var, coords={'x': var, 'aux': var})
```
Comment (SimonHeybrock) [inline python/tests/test_dataset.py:48]: This is a slicing test. Should we rather define `base` as in this line directly as `DataArray`? I believe the original test code is from the time when we did not have `DataArray`, thus the detour via `base['a']`.

File: python/tests/test_dataset.py [hunk 4]
```diff
@@ -385,7 +402,7 @@ def test_variable_histogram():
         hist.values, np.array([[1.0, 4.0, 1.0, 0.0], [0.0, 6.0, 0.0, 0.0]]))
 
 
-@pytest.mark.skip(reason="Requires unaligned dataset items")
+#@pytest.mark.skip(reason="Requires unaligned dataset items")
```
Comment (SimonHeybrock) [inline python/tests/test_dataset.py:405]: Remove instead of comment?

**Comment by SimonHeybrock** [comment]
github's diff now shows that some of the things you added in an earlier version of this PR are gone, is that intentional? Is github missing something?

**Comment by OwenArnold** [comment]
> github's diff now shows that some of the things you added in an earlier version of this PR are gone, is that intentional? Is github missing something?

It's nothing has been unexpectedly removed in recent work. Are you still getting a weird diff from github?

File: python/tests/test_data_array.py [hunk 5]
```diff
@@ -96,3 +198,33 @@ def test_reciprocal():
...
+    da = sc.DataArray(data=data, coords={'x': co})
+    assert not da.unaligned
+    da_r = sc.realign(
+        da, {'x': sc.Variable(['x'], values=np.array([0.0, 1.0, 3.0]))})
+    assert da_r.shape == [1, 2]
+    assert da_r.unaligned == da
+    assert not da_r.data
+    assert np.allclose(sc.histogram(da_r).values, np.array([0, 3]), atol=1e-9)
+    da.realign({'x': sc.Variable(['x'], values=np.array([0.0, 1.0, 3.0]))})
+    assert da.shape == [1, 2]
```
Comment (SimonHeybrock) [inline python/tests/test_data_array.py:221]: `realign` is tested in `test_realign.py`, any reason for putting a test here instead?

**Pull request merged**